### PR TITLE
Move input_size check into RunTime phrase of gru_unit_op and refine error message

### PR DESCRIPTION
**Pull request by Aurelius84**
PR types: Others

PR changes: Others

Describe:

+ Move `input_size` check in `RunTime` because  it can be -1 in CompileTime in `GRUUnitOp`
+ Refine Error Message of `GRUUnitOp`
+ Refine Error Message of `SelectInput/OutputOp`
+ Refine Error Message of `RNNMemoryHelperOp`

### CODE REVIEW

**Review by 1024er — COMMENTED** [review]
LGTM

File: paddle/fluid/operators/gru_unit_op.cc [hunk 1]
```diff
@@ -46,23 +41,45 @@ class GRUUnitOp : public framework::OperatorWithKernel {
...
-        input_size, frame_size * 3,
-        "The input_size must be 3 times of frame_size in GRUUnitOp.");
+    if (ctx->IsRuntime()) {
+      PADDLE_ENFORCE_EQ(input_size, frame_size * 3,
+                        platform::errors::InvalidArgument(
+                            "The second dimension of Input(Input) must be 3 "
+                            "times of frame_size in GRUUnitOp, but received %d "
+                            "(Input) vs %d (frame_size).",
+                            input_size, frame_size));
+    }
```
Comment (liym27) [inline paddle/fluid/operators/gru_unit_op.cc:51]: In compile time, checking `input_size` is necessary if `input_size` >= 0.

File: paddle/fluid/operators/gru_unit_op.cc [hunk 2]
```diff
@@ -166,23 +178,46 @@ class GRUUnitGradOp : public framework::OperatorWithKernel {
...
-        "The input_size must be 3 times of frame_size in GRUUnitOp.");
+    if (ctx->IsRuntime()) {
+      PADDLE_ENFORCE_EQ(
+          input_size, frame_size * 3,
+          platform::errors::InvalidArgument(
+              "The second dimension of Input(Input) must be 3 "
+              "times of frame_size in GRUUnitGradOp, but received %d "
+              "(Input) vs %d (frame_size).",
+              input_size, frame_size));
+    }
```
Comment (liym27) [inline paddle/fluid/operators/gru_unit_op.cc:189]: The same as above

File: paddle/fluid/operators/gru_unit_op.cc [hunk 3]
```diff
@@ -25,19 +25,14 @@ class GRUUnitOp : public framework::OperatorWithKernel {
...
-    PADDLE_ENFORCE(ctx->HasInput("Weight"),
-                   "Input(%s) of GRUUnitOp should not be null.", "Weight");
-    PADDLE_ENFORCE(ctx->HasOutput("Gate"),
-                   "Output(%s) of GRUUnitOp should not be null.", "Gate");
-    PADDLE_ENFORCE(ctx->HasOutput("ResetHiddenPrev"),
-                   "Output(%s) of GRUUnitOp should not be null.",
-                   "ResetHiddenPrev");
-    PADDLE_ENFORCE(ctx->HasOutput("Hidden"),
-                   "Output(%s) of GRUUnitOp should not be null.", "Hidden");
+    OP_INOUT_CHECK(ctx->HasInput("Input"), "Input", "Input", "GRUUnitOp");
```
Comment (chenwhql) [inline paddle/fluid/operators/gru_unit_op.cc]: 其实这个模板里已经有`operator`的，这里最后的描述还带着op可能有点奇怪，或者之后再改下
```
#define OP_INOUT_CHECK(__EXPR, __ROLE, __NAME, __OP_TYPE)                   \
  do {                                                                      \
    PADDLE_ENFORCE_EQ(__EXPR, true, paddle::platform::errors::NotFound(     \
                                        "No %s(%s) found for %s operator.", \
                                        __ROLE, __NAME, __OP_TYPE));        \
  } while (0)
```

**Review by chenwhql — DISMISSED** [review]

File: paddle/fluid/operators/gru_unit_op.cc (hunk 1 above)
Comment (Aurelius84) [reply paddle/fluid/operators/gru_unit_op.cc:51]: Good advice! Done.

File: paddle/fluid/operators/gru_unit_op.cc (hunk 3 above)
Comment (Aurelius84) [reply paddle/fluid/operators/gru_unit_op.cc]: Thanks, already removed suffix `op` in `__OP_TYPE`

File: paddle/fluid/operators/gru_unit_op.cc (hunk 2 above)
Comment (Aurelius84) [reply paddle/fluid/operators/gru_unit_op.cc:189]: Thanks，done.

**Review by liym27 — APPROVED** [review]
LGTM

**Review by 1024er — APPROVED** [review]

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Pull request merged**